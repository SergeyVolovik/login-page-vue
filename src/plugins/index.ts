import { createPinia } from 'pinia';

import PrimeVue from 'primevue/config';
import Aura from '@primevue/themes/aura';
import Button from 'primevue/button';
import Password from 'primevue/password';
import InputText from 'primevue/inputtext';
import Skeleton from 'primevue/skeleton';
import InputGroup from 'primevue/inputgroup';
import FloatLabel from 'primevue/floatlabel';
import Message from 'primevue/message';
import ProgressSpinner from 'primevue/progressspinner';
import ToastService from 'primevue/toastservice';
import Card from 'primevue/card';
import Rating from 'primevue/rating';
import Tag from 'primevue/tag';

import router from '@/router';

const PRIMEVUE_COMPONENTS = [
  { name: 'Button', component: Button },
  { name: 'Password', component: Password },
  { name: 'InputText', component: InputText },
  { name: 'Skeleton', component: Skeleton },
  { name: 'InputGroup', component: InputGroup },
  { name: 'FloatLabel', component: FloatLabel },
  { name: 'Message', component: Message },
  { name: 'ProgressSpinner', component: ProgressSpinner },
  { name: 'Card', component: Card },
  { name: 'Rating', component: Rating },
  { name: 'Tag', component: Tag }
];

// const CUSTOM_COMPONENTS = [{ name: 'Placeholder', component: Placeholder }];

export const registerPrimeVue = (app: any) => {
  PRIMEVUE_COMPONENTS.forEach(({ name, component }) => app.component(name, component));
};

// export const registerCustomComponents = (app: any) => {
//   CUSTOM_COMPONENTS.forEach(({ name, component }) => app.component(name, component));
// };

export const registerPlugins = (app: any) => {
  app
    .use(PrimeVue, {
      theme: {
        preset: Aura
      }
    })
    .use(ToastService)
    .use(router)
    .use(createPinia());

  registerPrimeVue(app);
};
