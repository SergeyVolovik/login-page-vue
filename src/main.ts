import { registerPlugins } from '@/plugins';
import { createApp } from 'vue';
import App from './App.vue';

import 'primeicons/primeicons.css';
import 'vue3-toastify/dist/index.css';
import './style.css';

const app = createApp(App);

registerPlugins(app);

app.mount('#app');
