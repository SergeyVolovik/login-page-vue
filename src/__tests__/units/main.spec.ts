import App from '@/App.vue';
import { registerPlugins } from '@/plugins';
import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest';
import { createApp } from 'vue';

import './../../main';

describe('main.ts', () => {
  let appContainer: any;

  beforeEach(() => {
    appContainer = document.createElement('div');
    appContainer.id = 'app';
    document.body.appendChild(appContainer);
  });

  afterEach(() => {
    document.body.removeChild(appContainer);
    appContainer = null;
    vi.clearAllMocks();
    vi.resetAllMocks();
  });

  it('calls registerPlugins with the app instance', () => {
    const registerPluginsMock = vi.fn();
    const app = createApp(App);
    registerPluginsMock(app);
    expect(registerPluginsMock).toHaveBeenCalledWith(app);
  });

  it('mounts the app instance to the #app element', () => {
    const app = createApp(App);
    registerPlugins(app);
    app.mount('#app');
    expect(document.getElementById('app')).toBeTruthy();
  });
});
