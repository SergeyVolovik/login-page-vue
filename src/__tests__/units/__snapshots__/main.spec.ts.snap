// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > renders 1`] = `""`;
